datatypes/ring_programs.py

```diff
@@ -130,10 +136,10 @@ def _combine_single_sections(cls, *args,
 
         timeBases = [a.timebase for a in args]
         try:
-            assrt.equal_arrays(*timeBases, msg = 'Attempting to combine '
-                                                 + 'sections with different '
-                                                 + 'timebases',
-                                                 exception = excpt.InputError)
+            assrt.equal_arrays(*timeBases, msg='Attempting to combine '
+                               + 'sections with different '
+                               + 'timebases',
+                               exception=excpt.InputError)
         except:
             turns = len([t for t in timeBases if t == 'by_turn']) == 0
             times = len([t for t in timeBases if t == 'by_time']) == 0
```

Comment:
This bit confuses me.  What's it for?

Reply:
This is for you to answer 😛 

I didn't touch anything in there except the `n_turns` in `preprocess` and the `cycle_time` part, the rest is only autopep8

Reply:
huh...